hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -328,7 +328,9 @@ public boolean readFrom(ByteBuffer src) {
                 // we don't have even the frame length ready
                 return false;
             }
-            frameLength = Bits.readIntL(src.array(), src.position());
+            frameLength = Bits.readIntL(src);
```

Comment:
this would involve storing the older order and resetting it after the 'getInt' completes. I think that is more complex than the current approach.